apps/src/lab2/views/LabViewsRenderer.tsx

```diff
@@ -38,27 +41,60 @@ const LabViewsRenderer: React.FunctionComponent = () => {
   );
 
   const isViewingExemplar = getAppOptionsViewingExemplar();
-
-  const capitalizedLessonBackground = useAppSelector(
-    state =>
-      capitalizeFirstLetter(
-        getCurrentLesson(state)?.background || 'light'
-      ) as Theme
+  const lesson = useAppSelector(state => getCurrentLesson(state));
+  const {signInState} = useAppSelector(state => state.currentUser);
+  const dispatch = useAppDispatch();
+
+  // We only use the global user preference for theme if the current lesson has
+  // at least one python lab level or the current level is a python lab level.
+  const useThemeUserPreference = useMemo(
+    () =>
+      levelProperties?.appName === 'pythonlab' ||
+      lesson?.levels.some((level: Level) => level.app === 'pythonlab'),
+    [lesson?.levels, levelProperties?.appName]
   );
 
   // Set the theme for the current app.
   const {setTheme} = useTheme();
   useEffect(() => {
     if (currentAppName) {
       const supportedThemes = lab2EntryPoints[currentAppName]?.themes;
-
-      if (supportedThemes.includes(capitalizedLessonBackground)) {
-        setTheme(capitalizedLessonBackground);
+      dispatch(setIsLoadingTheme(true));
+
+      const setThemeHelper = () => {
```

Comment:
Not sure if it's worthwhile here, but we could extract some of this business logic into a "helper" hook in its own file?